react_ujs/index.js

```diff
@@ -80,6 +82,11 @@ var ReactRailsUJS = {
     return ReactDOMServer[renderFunction](element)
   },
 
+  // Render `component` using the specified `renderFunction` from `react-dom`.
+  // Override this function to render components in a custom way.
+  // function signature: ("hydrate" | "render", component, node, props)
+  renderComponent: renderWithReactDOM,
```

Comment:
The comment and pattern here intends to mirror `getConstructor`. Let me know if

1. the function signature can be better represented
1. if the type of `renderFunctionName` (`"hydrate" | "render"`) should be made more generic, i.e. `string`
1. export the `renderWithReactDOM` function or assign it in a constant, e.g. `ReactRailsUJS.DEFAULT_RENDER_COMPONENT_FUNCTION`

Reference:
https://github.com/reactjs/react-rails/blob/d5da11129459cd75fd003c75319b1f7440c37322/react_ujs/index.js#L63-L66